lib/api/controllers/adminController.js

```diff
@@ -52,6 +56,37 @@ class AdminController extends BaseController {
     ]);
   }
 
+  decryptSecrets (request) {
+    assertHasBody(request);
+    assertBodyHasAttribute(request, 'secrets');
+    assertBodyHasAttribute(request, 'vaultKey');
+
+    const vaultKey = request.input.body.vaultKey,
+      secrets = request.input.body.secrets,
+      vault = new Vault();
```

Comment:
You can use the vault instantiated by Kuzzle instead I think `this.kuzzle.vault`